src/typechecker.ts

```diff
@@ -332,6 +435,152 @@ function isConsumedAsCompletion(expr: Expr, path: PathItem[]) {
   return false;
 }
 
+// returns a boolean to indicate whether this line can return an abrupt completion, or null to indicate we should stop caring
```

Comment:
I don't really like that. Can't we make an enum for it?

Reply:
Then we couldn't use `||=`.

Reply:
just implement `Functor` 😛

Reply:
no